### PR TITLE
Update schnorrkel with versioning

### PR DESCRIPTION
**Pull request by NikVolf**
This is a clone of #5336 (reviewed already)

### CODE REVIEW

File: primitives/core/src/sr25519.rs [hunk 1]
```diff
@@ -566,6 +565,20 @@ impl Pair {
 		let kp = mini_key.expand_to_keypair(ExpansionMode::Ed25519);
 		(Pair(kp), mini_key.to_bytes())
 	}
+
+	/// Verify a signature on a message. Returns true if the signature is good.
```
Comment (bkchr) [inline primitives/core/src/sr25519.rs]:
```suggestion
	/// Verify a signature on a message. Returns `true` if the signature is good.
```

**Comment by NikVolf** [comment]
> You tested the old version on Kusama already?

Yes, but I am also testing exactly this pr just in case

**Comment by bkchr** [comment]
What you think, should we add a test that building a block with an old signature fails?

**Comment by NikVolf** [comment]
> What you think, should we add a test that building a block with an old signature fails?

Yeah, we should

@burdges said he yanked old libraries who can produce such signatures though, so we need to figure out what commit from git to use

**Comment by NikVolf** [comment]
@gavofyork will add a test here, it is not so quick (requires tackling with old version of schnorrkel)

**Pull request merged**